### PR TITLE
Instructions to set up a development environment

### PR DESCRIPTION
**Pull request by tamasgal**
This is a tiny commit which adds a `requirements-dev.txt` containing the packages needed to run the suite and a short description how to set up a working environment.

I always forget which packages are needed for the test suite and I run `pytest` a few times. This will make it a bit easier :wink:

### CODE REVIEW

**Comment by tamasgal** [comment]
Yes, that I understood. My instructions are indeed aimed for what you have described in your last paragraph, I just skipped that it's cloned via git as I assumed it is anyways.

Following that, maybe a tiny change of the line `cd /path/to/...` to `git clone ... && cd ..` is maybe enough, or what do you think?

The main idea was to make it a bit simpler to get started with a green test suite.

File: README.rst [hunk 1]
```diff
@@ -77,6 +77,22 @@ Optional dependencies:
...
+The ``requirements-dev.txt`` file contains the packages needed to run the test
+suite. Here is an example how to set up a development environment:
+
+.. code-block:: bash
+
+    cd /path/to/your/cloned/repo/or/fork/of/uproot
+    python -m venv venv  # create a virtual environment in the folder venv
+    . venv/bin/activate  # activate the Python environment
+    pip install -e .     # this will install uproot in "editable mode"
+    pip install -r requirements-dev.txt  # install the dev-requirements
```
Comment (chrisburr) [inline README.rst]: Personally I prefer to add a `extras_require` section in `setup.py` called `dev` ([example](https://stackoverflow.com/a/28842733/2685230)) so you can do `pip install -e .[dev]`.

Comment (tamasgal) [reply README.rst]: Yeah, why not :wink:

Comment (tamasgal) [reply README.rst]: I see that there is already `tests_require`, but not sure how to use that in conjuction with `pip`.

Comment (tamasgal) [reply README.rst]: It seems that it's simply `pip install -e ".[testing]"`

Comment (tamasgal) [reply README.rst]: However, it does not install anything else other than the main requirements :thinking: 

```
░ tgal@staticbox:uproot  requirements-dev $● uproot 
░ 14:35:33 > pip install -e ".[testing]"
Obtaining file:///home/tgal/Dev/uproot
  WARNING: uproot 3.11.0 does not provide the extra 'testing'
Collecting numpy>=1.13.1 (from uproot==3.11.0)
  Using cached https://files.pythonhosted.org/packages/41/38/b278d96baebc6a4818cfd9c0fb6f0e62013d5b87374bcf0f14a0e9b83ed5/numpy-1.18.1-cp38-cp38-manylinux1_x86_64.whl
Collecting awkward<1.0,>=0.12.0 (from uproot==3.11.0)
  Using cached https://files.pythonhosted.org/packages/35/f3/d86d52275cfe131692c89558a22187e40cf111485154affac74d7509ee43/awkward-0.12.19-py2.py3-none-any.whl
Collecting uproot-methods>=0.7.0 (from uproot==3.11.0)
  Using cached https://files.pythonhosted.org/packages/3c/39/7867f279a8ca8a67ae8ca56c371dda6cc8e05dc01ac614f05586c857dfe2/uproot_methods-0.7.2-py2.py3-none-any.whl
Collecting cachetools (from uproot==3.11.0)
  Using cached https://files.pythonhosted.org/packages/08/6a/abf83cb951617793fd49c98cb9456860f5df66ff89883c8660aa0672d425/cachetools-4.0.0-py3-none-any.whl
Installing collected packages: numpy, awkward, uproot-methods, cachetools, uproot
  Running setup.py develop for uproot
Successfully installed awkward-0.12.19 cachetools-4.0.0 numpy-1.18.1 uproot uproot-methods-0.7.2
WARNING: You are using pip version 19.2.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

Comment (tamasgal) [reply README.rst]: Ah, I mixed that up... It should be under `extras_require` under the `testing` keyword.

Is `tests_require` used somewhere in your CI?

**Comment by tamasgal** [comment]
I just updated the `setup.py` to use `extras_require` with a `"testing"` entry. Not sure though if the `tests_require` is used somewhere in your CI or so.

**Comment by jpivarski** [comment]
The `tests_require` is not explicitly relied upon in CI.

I like this solution using `extras_require`! It's good to know that Python has a built-in concept of "optional dependencies."

**Comment by tamasgal** [comment]
Alright, ready for merge then. Thanks!

**Pull request merged**